### PR TITLE
Data: restore explicit data removal of private data in cleanData.

### PR DESCRIPTION
**Pull request by rwaldron**
Fixes gh-2127

### CODE REVIEW

File: src/manipulation.js [hunk 1]
```diff
@@ -306,6 +306,7 @@ jQuery.extend({
 					}
 				}
 				delete data.events;
+				delete elem[dataPriv.expando];
```
Comment (gibson042) [inline src/manipulation.js]: Missing spaces around the property. Also, can we drop the preceding `delete data.events`?

Comment (rwaldron) [inline src/manipulation.js]: > Missing spaces around the property. 

I see that the circle of life is complete ;)

File: src/manipulation.js [hunk 2]
```diff
@@ -305,7 +305,7 @@ jQuery.extend({
 						}
 					}
 				}
-				delete data.events;
+				delete elem[ dataPriv.expando ];
```
Comment (rwaldron) [inline src/manipulation.js]: cc @gibson042

File: src/manipulation.js (hunk 1 above)
Comment (gibson042) [inline src/manipulation.js]: You said it. :bow:

**Pull request closed without merging**